editor/filesystem_dock.cpp

```diff
@@ -2048,7 +2048,7 @@ Vector<String> FileSystemDock::_remove_self_included_paths(Vector<String> select
 }
 
 void FileSystemDock::_tree_rmb_option(int p_option) {
-	Vector<String> selected_strings = _tree_get_selected(false);
+	Vector<String> selected_strings = _tree_get_selected(false, true);
```

Comment:
What does this change fix?

Reply:
The issue described here: https://github.com/godotengine/godot/issues/82543#issuecomment-1751042987
#82045 made it so that currently focused but unselected item in the tree wouldn't be included in the selected item list, but this broke the renaming behavior because visually renamed item (tree->get_selected()) and the path being renamed (to_rename.path) became different. https://github.com/godotengine/godot/blob/42425baa59956dc9d1e22341fe5e5d7f8fad5067/editor/filesystem_dock.cpp#L2232-L2235

So, I added an optional parameter to also include the unselected focused item to the list, to restore the old behavior. But I noticed that making this change at the top of _tree_rmb_option() wouldn't be the correct fix, because we want this behavior only for renaming. So, maybe I should add another call to  _tree_get_selected() with second argument as true inside the FILE_RENAME case, and revert the one in _tree_rmb_option(). What do you think?

Reply:
Right, this function confused me a bit.

I think it's good here (right-clicking an item has the same problem actually), but I agree that it should only affect FILE_RENAME. You can add another `case` that overwrites selected strings (although it kind of sucks to call `_tree_get_selected()` twice; maybe it can be done better):
```C++
case FILE_RENAME: {
	selected_strings = _tree_get_selected(false, true);
	[[fallthrough]];
}
default: {
	_file_option(p_option, selected_strings);
} break;
```